docs/gadgetController.rst

```diff
@@ -0,0 +1,122 @@
+.. _gadgetController:
+
+Gadget Controller Interface Reference
+=====================================
+
+The `Gadget Controller interface <https://developer.amazon.com/docs/gadget-skills/gadgetcontroller-interface-reference.html>`_ enables your skill to control Echo Buttons. This interface works with compatible Amazon Echo devices only. With the Gadget Controller interface, you can send animations to illuminate the buttons with different colors in a specific order.
+
+With Voxa, you can implement this interface like this:
+
+.. code-block:: javascript
+
+  const voxa = require('voxa');
+  const { GadgetController, TRIGGER_EVENT_ENUM } = voxa.alexa;
+
+  skill.onIntent('GameEngine.InputHandlerEvent', (voxaEvent) => {
+    // REMEMBER TO SAVE THE VALUE originatingRequestId in your model
+    voxaEvent.model.originatingRequestId = voxaEvent.request.originatingRequestId;
+
+    const gameEvents = voxaEvent.request.events[0] || [];
+    const inputEvents = _(gameEvents.inputEvents)
+      .groupBy('gadgetId')
+      .map(value => value[0])
+      .value();
+
+    const directives = [];
+    let customId = 0;
+
+    _.forEach(inputEvents, (gadgetEvent) => {
+      customId += 1;
+      const id = `g${customId}`;
+
+      if (!_.includes(voxaEvent.model.buttons, id)) {
+        const buttonIndex = _.size(voxaEvent.model.buttons);
+        const targetGadgets = [gadgetEvent.gadgetId];
+
+        _.set(voxaEvent.model, `buttonIds.${id}`, gadgetEvent.gadgetId);
+
+        voxaEvent.model.buttons = [];
+        voxaEvent.model.buttons.push(id);
+
+        const triggerEventTimeMs = 0;
+        const gadgetController = new GadgetController();
+        const animationBuilder = GadgetController.getAnimationsBuilder();
+        const sequenceBuilder = GadgetController.getSequenceBuilder();
+
+        sequenceBuilder
+          .duration(1000)
+          .blend(false)
+          .color(COLORS[buttonIndex].dark);
+
+        animationBuilder
+          .repeat(100)
+          .targetLights(['1'])
+          .sequence([sequenceBuilder]);
+
+        directives.push(gadgetController
+          .setAnimations(animationBuilder)
+          .setTriggerEvent(TRIGGER_EVENT_ENUM.NONE)
+          .setLight(targetGadgets, triggerEventTimeMs));
+
+        const otherAnimationBuilder = GadgetController.getAnimationsBuilder();
+        const otherSequenceBuilder = GadgetController.getSequenceBuilder();
+
+        otherSequenceBuilder
+          .duration(500)
+          .blend(false)
+          .color(COLORS[buttonIndex].hex);
+
+        otherAnimationBuilder
+          .repeat(1)
+          .targetLights(['1'])
+          .sequence([otherSequenceBuilder.build()]);
+
+        directives.push(gadgetController
+          .setAnimations(otherAnimationBuilder.build())
+          .setTriggerEvent(TRIGGER_EVENT_ENUM.BUTTON_DOWN)
+          .setLight(targetGadgets, triggerEventTimeMs));
+      }
+    });
+
+    return { tell: 'Buttons.Next', directives, to: 'entry' };
```

Comment:
I'm not sure if we can update it in this particular case, because I'm sending 2 different directives:

```
GameController.Setlight
GameEngine.StartInputHandler
```

Reply:
something like 

```js
{
alexaSetLightDirective: ...
alexaGameEngineStartInputHandler: ...
}
```

??

Reply:
Nevermind, it is doable